### PR TITLE
Ensure election managers and system admins can set date and time for all machines

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Fixes: #4329 

Currently, different apps allow different users to set the date and time. This PR enables setting the date and time for election managers and system administrators in every app.

Also spruces up the design of the modal for setting the date and time a bit. 

## Demo Video or Screenshot
VxAdmin
https://github.com/votingworks/vxsuite/assets/530106/3c8522d1-da91-4d8f-929a-63b68843d6a9

VxCentralScan

https://github.com/votingworks/vxsuite/assets/530106/2ddbcc1c-6ffc-434f-a791-45d7d089e655

VxScan

https://github.com/votingworks/vxsuite/assets/530106/b6be070b-5eb2-483f-a60a-ebe716ec8e0b

VxMark/VxMarkScan

https://github.com/votingworks/vxsuite/assets/530106/1224bdb0-6e74-4b6c-ac91-a883822e612a

## Testing Plan
Manual testing
Update automated tests (TODO)

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/frontend/src/components/navigation_screen.tsx [hunk 1]
```diff
@@ -55,10 +55,7 @@ const ELECTION_MANAGER_NAV_ITEMS: readonly NavItem[] = _.compact([
     BooleanEnvironmentVariableName.WRITE_IN_ADJUDICATION
   ) && { label: 'Write-Ins', routerPath: routerPaths.writeIns },
   { label: 'Reports', routerPath: routerPaths.reports },
-  isFeatureFlagEnabled(BooleanEnvironmentVariableName.LIVECHECK) && {
-    label: 'System',
-    routerPath: routerPaths.system,
-  },
```
Comment (eventualbuddha) [inline apps/admin/frontend/src/components/navigation_screen.tsx:61]: LiveCheck removal intentional?

File: apps/scan/frontend/src/app_root.tsx [hunk 2]
```diff
@@ -184,33 +187,35 @@ export function AppRoot({
...
-                    time: Date.now(),
-                  })
-              : undefined
-          }
-          isMachineConfigured={Boolean(electionDefinition)}
-          usbDriveStatus={usbDrive}
-          additionalButtons={additionalButtons}
-        />
-      </ScreenMainCenterChild>
+      <Screen title="System Administrator">
```
Comment (eventualbuddha) [inline apps/scan/frontend/src/app_root.tsx:190]: Should this likewise be moved into its own file?

**Review by eventualbuddha — APPROVED** [review]
LGTM with one question about LiveCheck.

File: apps/admin/frontend/src/components/navigation_screen.tsx (hunk 1 above)
Comment (jonahkagan) [reply apps/admin/frontend/src/components/navigation_screen.tsx:61]: The LiveCheck logic is already in the settings screen that I replaced this with. So it's just consolidated now

File: apps/scan/frontend/src/app_root.tsx (hunk 2 above)
Comment (jonahkagan) [reply apps/scan/frontend/src/app_root.tsx:190]: Sure, can do

**Pull request merged**